### ISSUE TITLE
Sleep on MinGW has only 1ms precision

### ISSUE THREAD

**Issue by Bromeon** [SFML/SFML#1946]
## Subject of the issue

As reported in #1936, `std::this_thread::sleep()` seems to have only millisecond (not microsecond) precision on Windows with MinGW.

This also affects `sf::sleep()` which is implemented using the afore-mentioned standard method. A platform differentiation could work around the limitation on MinGW.

## Expected behavior

Allow sleep times in the order of 100us, ideally better.

## Actual behavior

Sleep has 1ms precision.

**Comment by ChrisThrasher** [comment]
The following test should be able to reproduce the bug:
```cpp
const sf::Clock clock;
std::this_thread::sleep_for(std::chrono::microseconds(100));
CHECK(clock.getElapsedTime() > sf::Time::Zero);
```
The failed test should report that the elapsed time is zero.

**Comment by vittorioromeo** [comment]
This should be reported (and hopefully fixed) upstream: https://sourceforge.net/p/mingw-w64/bugs/

**Comment by vittorioromeo** [comment]
Possibly related? https://stackoverflow.com/questions/12523122/what-is-glibcxx-use-nanosleep-all-about

**Comment by vittorioromeo** [comment]
I don't think this is an issue with MinGW. I think this is an issue with `libstdc++`. See https://github.com/gcc-mirror/gcc/blob/master/libstdc++-v3/include/bits/this_thread_sleep.h 

```cpp
    /// this_thread::sleep_for
    template<typename _Rep, typename _Period>
      inline void
      sleep_for(const chrono::duration<_Rep, _Period>& __rtime)
      {
	if (__rtime <= __rtime.zero())
	  return;
	auto __s = chrono::duration_cast<chrono::seconds>(__rtime);
	auto __ns = chrono::duration_cast<chrono::nanoseconds>(__rtime - __s);
#ifdef _GLIBCXX_USE_NANOSLEEP
	struct ::timespec __ts =
	  {
	    static_cast<std::time_t>(__s.count()),
	    static_cast<long>(__ns.count())
	  };
	while (::nanosleep(&__ts, &__ts) == -1 && errno == EINTR)
	  { }
#else
	__sleep_for(__s, __ns);
#endif
      }
```

MinGW does provide `nanosleep` implementation AFAICS, and `libstdc++` should detect it and use it. Have we noticed this issue only in our CI? Do we have an outdated version of MinGW + `libstdc++`? I'm gonna test it out on my MSYS2 installation.

**Comment by vittorioromeo** [comment]
Addressed by #1965.
